src/P3_processes.jl

```diff
@@ -227,3 +270,254 @@ function ρ′_rim_P3(Rᵢ)
+    ice_bounds = integral_bounds(state, logλ; p = 0.00001)
+    mm = FT(1e-3)
+    bounds_c = FT[0; 0.01mm; 0.1mm; 1mm; 10mm; 100mm; 1]  # TODO: Replace by quantiles method
+    bounds_r = FT[0; 0.01mm; 0.1mm; 1mm; 10mm; 100mm; 1]  # TODO: Replace by quantiles method
+    segbuf_c = QGK.quadgk_segbuf(n_c, bounds_c...)[3]
+    segbuf_r = QGK.quadgk_segbuf(n_r, bounds_r...)[3]
+    segbuf_ice = QGK.quadgk_segbuf(n_i, ice_bounds...)[3]
+
+    # Integrand components
+    # NOTE: We assume collision efficiency, shape (spherical), and terminal velocity is the 
+    #   same for cloud and precipitating liquid particles ⟹ same volumetric collision rate, ∂ₜV
+    ∂ₜV = volumetric_collision_rate_integrand(state, vel, ρₐ)  # ∂ₜV(Dᵢ, Dₗ)
+    ρ′_rim = compute_local_rime_density(state, vel, ρₐ, T)  # ρ′_rim(Dᵢ, Dₗ)
+    ∂ₜM_max = compute_max_freeze_rate(state, aps, tps, vel, ρₐ, T)  # ∂ₜM_max(Dᵢ)
+
+    cloud_integrals = get_liquid_integrals(n_c, ∂ₜV, m_liq, ρ′_rim, bounds_c; eval_segbuf = segbuf_c)  # (∂ₜN_c_col, ∂ₜM_c_col, ∂ₜB_c_col)
+    rain_integrals = get_liquid_integrals(n_r, ∂ₜV, m_liq, ρ′_rim, bounds_r; eval_segbuf = segbuf_r)  # (∂ₜN_r_col, ∂ₜM_r_col, ∂ₜB_r_col)
+
+    return ∫liquid_ice_collisions(
+        n_i, ∂ₜM_max, cloud_integrals, rain_integrals, ice_bounds; eval_segbuf = segbuf_ice,
+    )
+end
+
+"""
+    bulk_liquid_ice_collision_sources(
+        params, logλ, L_ice, F_rim, ρ_rim,
+        psd_c, psd_r, L_c, N_c, L_r, N_r,
+        aps, tps, vel, ρₐ, T,
+    )
+
+Computes the bulk rates for ice and liquid particle collisions.
+
+# Arguments
+- `params`: the [`CMP.ParametersP3`](@ref)
+- `logλ`: the log of the slope parameter [log(1/m)]
+- `L_ice`: ice water content [kg/m³]
+- `F_rim`: riming fraction
+- `ρ_rim`: rime density [kg/m³]
+- `psd_c`: a [`CMP.CloudParticlePDF_SB2006`](@ref)
+- `psd_r`: a [`CMP.RainParticlePDF_SB2006`](@ref)
+- `L_c`: cloud liquid water content [kg/m³]
+- `N_c`: cloud liquid water number concentration [1/m³]
+- `L_r`: rain water content [kg/m³]
+- `N_r`: rain number concentration [1/m³]
+- `aps`: [`CMP.AirProperties`](@ref)
+- `tps`: thermodynamics parameters
+- `vel`: the velocity parameterization, e.g. [`CMP.Chen2022VelType`](@ref)
+- `ρₐ`: air density [kg/m³]
+- `T`: temperature [K]
+
+# Returns
+A `NamedTuple` of `(; ∂ₜq_c, ∂ₜq_r, ∂ₜN_c, ∂ₜN_r, ∂ₜL_rim, ∂ₜL_ice, ∂ₜB_rim)`, where:
+1. `∂ₜq_c`: cloud liquid water content tendency [kg/kg/s]
+2. `∂ₜq_r`: rain water content tendency [kg/kg/s]
+3. `∂ₜN_c`: cloud number concentration tendency [1/m³/s]
+4. `∂ₜN_r`: rain number concentration tendency [1/m³/s]
+5. `∂ₜL_rim`: riming mass tendency [kg/m³/s]
+6. `∂ₜL_ice`: ice water content tendency [kg/m³/s]
+7. `∂ₜB_rim`: rime volume tendency [m³/m³/s]
+"""
+function bulk_liquid_ice_collision_sources(
+    params, logλ, L_ice, F_rim, ρ_rim,
+    psd_c, psd_r, L_c, N_c, L_r, N_r,
+    aps, tps, vel, ρₐ, T,
+)
+    FT = eltype(params)
+    τ_wet = FT(100)  # [s]  # TODO: Externalize this parameter
```

Comment:
Would be nice to grab them from ClimaParams

Reply:
Proposed in https://github.com/CliMA/ClimaParams.jl/pull/236